### PR TITLE
doc/md: update MaxRuneCount example

### PR DESCRIPTION
**Pull request by giautm**
Update MaxRuneCount example with entsql.Annotation. 

https://github.com/ent/ent/issues/1637
https://github.com/ent/ent/issues/1856
https://github.com/ent/ent/issues/1914

### CODE REVIEW

**Review by maxifom — APPROVED** [review]

**Review by masseelch — COMMENTED** [review]
I'd add a note why this is necessary (`MaxLen` does count bytes which may result in unexpected behaviour for multibyte char collation like russian, chinese, smileys, ...). Other than that LGTM.

File: doc/md/schema-fields.md [hunk 1]
```diff
@@ -362,9 +362,14 @@ func (Group) Fields() []ent.Field {
 }
 ```
 
-Here is another example for writing a reusable validator:
+`MaxLen` does count bytes which may result in unexpected behavior for multibyte char collation like Russian, Chinese, Japanese, Vietnamese, smileys, ... Here is another example for writing a reusable validator:
```
Comment (masseelch) [inline doc/md/schema-fields.md:365]:
```suggestion
Here is another example for writing a reusable validator:
```

I'd add the part about `MaxLen` counting bytes to the Built-in Validators section and tell the user to look at this example for a "fix".

File: doc/md/schema-fields.md [hunk 2]
```diff
@@ -376,8 +381,16 @@ func MaxRuneCount(maxLen int) func(s string) error {
 }
 
 field.String("name").
+	// Change the underlying data type to varchar(10).
```
Comment (masseelch) [inline doc/md/schema-fields.md:384]:
```suggestion
	// If using an SQL-database: change the underlying data type to varchar(10).
```

File: doc/md/schema-fields.md [hunk 3]
```diff
@@ -376,8 +381,16 @@ func MaxRuneCount(maxLen int) func(s string) error {
...
+	Annotations(entsql.Annotation{
+		Size: 10,
+	}).
 	Validate(MaxRuneCount(10))
 field.String("nickname").
+	// Change the underlying data type to varchar(20).
```
Comment (masseelch) [inline doc/md/schema-fields.md:390]:
```suggestion
	//  If using an SQL-database: change the underlying data type to varchar(20).
```

**Review by masseelch — COMMENTED** [review]
:+1:

**Review by a8m — APPROVED** [review]
Thanks, @giautm for the contribution, and thanks @masseelch and @maxifom for the review.

**Pull request merged**